src/Symfony/Component/Security/Core/Authentication/Token/AbstractToken.php

```diff
@@ -33,7 +33,7 @@
     /**
      * Constructor.
      *
-     * @param RoleInterface[] $roles An array of roles
+     * @param RoleInterface[]|string[] $roles An array of roles
```

Comment:
There are already some debates about this notation, being incomplete. See https://github.com/symfony/symfony/pull/13146#discussion_r22309134 and other related issues.

As @dosten said in one of them:
> `getRoles` can return an array of strings, Role instances or a mix of both, the phpdoc must reflect that.